core/typedefs.h

```diff
@@ -315,4 +315,6 @@ struct _GlobalLock {
 #define _PRINTF_FORMAT_ATTRIBUTE_2_3
 #endif
 
+#define CAST_INT_TO_UCHAR_PTR(ptr) ((uint8_t *)(uintptr_t)(ptr))
```

Comment:
See discussion with @reduz and @hpvb here: http://godot.eska.me/irc-logs/meeting/2019-03-02.log (from 22:20).

Reply:
I've read it and I must say that first the signature of the function is bad. I would expect an unsigned int as for the offset. Second, I still believe that ```(uint_8 *)0 + some_offsets``` is uglier that usage of ```uintptr_t``` and the casting to ```char *```. Casting to ```char *``` is intentional for aliasing reasons (aliases with a random pointer to e.g. structure). And ```uintptr_t``` is designed to represent a pointer value.

Reply:
Good, I've just added a comment ;)